### PR TITLE
Adjust init function for new network

### PR DESCRIPTION
**Pull request by charlesjhongc**
~According to Tokenlon contract version control policy, remove version stored in contract. Protect `initialize` function being called twice using `Initializable` contract from OZ. Set operator in `initialize` function.~

Adjust init function for new network deployment. May change design in the future.

### CODE REVIEW

**Comment by NIC619** [comment]
Ugly but simple and clear approach:
```solidity
function initialize(xxx) external {
        if (chainID = 1) { // Mainnet
               ...
        } else if (chainID = 42161) { // Arbitrum
               ...
        }
}
```

**Comment by pilagod** [comment]
Another thought is to make initialize idempotent, like

```solidity
function initialize(address _operator) public {
    if (operator == address(0)) {
        operator = _operator
    }
}
```

Instead of constraining initialize can only be called once, we can ensure that it can only take effect once, no matter how many calls.

By this method, we can highlight all required variables that the contract must to have to work in initialize. And new required variable will just append to it like

```solidity
function initialize(address _operator, address _emergencyRecipient) public {
    if (operator == address(0)) {
        operator = _operator
    }
    if (emergencyRecipient == address(0)) {
        emergencyRecipient = _emergencyRecipient
    }
}
```

This can make our contract resilient enough to deploy at any time, any network.

**Comment by charlesjhongc** [comment]
Force pushed to squash commits.

File: contracts/test/UserProxy.t.sol [hunk 1]
```diff
@@ -67,15 +54,19 @@ contract UserProxyTest is Test {
      *********************************/
 
     function testCannotInitializeToZeroAddress() public {
-        vm.expectRevert("UserProxy: _limitOrderAddr should not be 0");
-        userProxy.initialize(address(0));
+        UserProxy us = new UserProxy();
```
Comment (pilagod) [inline contracts/test/UserProxy.t.sol:57]: nit: Use `up` for consistency

**Review by pilagod — APPROVED** [review]

File: contracts/test/UserProxy.t.sol (hunk 1 above)
Comment (charlesjhongc) [reply contracts/test/UserProxy.t.sol]: Forgot this one. Fixed in cc8abb4.

**Review by NIC619 — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

**Pull request merged**